src/hooks/useFetch.js

```diff
@@ -51,15 +52,27 @@ export const useFetch = (url, options = {}, timeout = CONNECTION_TIMEOUT_PERIOD)
             if (!responseData.length) throw new Error(ERROR_CODES.RESPONSE_NO_RECORDS);
             if (isMounted.current) setData(responseData);
           } else {
+            const connectionError = new Error(ERROR_CODES.CONNECTION_FAILURE);
+            const authorizationError = new Error(ERROR_CODES.INVALID_PASSWORD);
             switch (status) {
               case 400:
-              default:
-                throw new Error(ERROR_CODES.CONNECTION_FAILURE);
+                throw connectionError;
               case 401:
-                throw new Error(ERROR_CODES.INVALID_PASSWORD);
+                throw authorizationError;
+              default:
+                bugsnagClient.notify(connectionError, content => {
+                  content.requestUrl = url;
+                  content.requestHeaders = headers;
+                  content.responseStatus = status;
+                });
+                throw connectionError;
```

Comment:
sorry! im confused why we're snagging this one twice, one with the status one without. If we just throw the connection error and add the status into the catch snag, would that work?

Reply:
Unfortunately the status is inside the `try`, so can't access it. It's not a biggie, removed in 63fcda1.